### PR TITLE
Feature: Content app sidebar redesign

### LINKED ISSUE #1385: Reordering content navigation tree not working properly
Trying to reorder the navigation tree for content items by dragging and dropping is inaccurate and does not work as expected. The dragged item jumps to the top or to the a different position than intended. 
See behavior in this video:

https://www.loom.com/share/ad66bae0ae044752a9ff53752f36f0e9

### LINKED ISSUE #2115: Content: App Sidebar Revamp
To increase the contrast between the sidebar and the main content area, we are making the content app sidebar dark. In addition to this, we are making it match, the new material UI theming of Zesty. This same style of App sidebar will also reflect in Media and in Schema to ensure consistency across the system. 

image

### PR DESCRIPTION
**Pull request by theofficialnar**
- Updated content app sidebar
- Closes #1385 
- Closes #2115 

### Note:
**This PR is dependent on https://github.com/zesty-io/material/pull/78 for proper styling**

### Preview
[content-app-sidebar.webm](https://github.com/zesty-io/manager-ui/assets/28705606/d2597b42-ba4b-4556-b745-4cab6d51f1fa)

### CODE REVIEW

**Review by zcolah — CHANGES_REQUESTED** [review]
@agalin920 
1. The sidebar resizing currently has a lot of lag and is not a smooth experience. It responds late sometimes, bounces back to a wider size when being made small, and sometimes skips frames. This does not happen on each attempt. It happens on one out of every 3 attempts on average. Please refer to Slack's and Clickup's sidebar for examples of smooth resizing.

**Comment by zcolah** [comment]
@shrunyan could we add the tooltip helper text on hover functionality to the scope of this project or should it be a different project? You can see the tooltip helper text bubbles here: https://www.figma.com/file/zOI7oSH3fG1XDmmPzsP6PX/App-Navigation?type=design&node-id=545%3A202423&mode=design&t=24AXSMD60gHgmc5T-1

**Review by zcolah — CHANGES_REQUESTED** [review]
@theofficialnar 
1. The flow of when a user hovers over a dataset or multi page model list item and clicks on the plus button to create an item does not follow the logic defined here in Figma: https://www.figma.com/file/zOI7oSH3fG1XDmmPzsP6PX/App-Navigation?type=design&node-id=52%3A71151&mode=design&t=24AXSMD60gHgmc5T-1

Instead of directly leading to the create item page, it instead opens a modal that is not part of the flow (see screenshot below): 
CleanShot 2023-07-31 at 16 43 40@2x


2. When a user searches for an item successfully or unsuccessfully we want to hide the dashboard and releases tabs. Unfortunately at the moment this is not the case for when a user searches successfully. 

Figma Reference to see defined design spec for search results: https://www.figma.com/file/zOI7oSH3fG1XDmmPzsP6PX/App-Navigation?type=design&node-id=52%3A70317&mode=design&t=24AXSMD60gHgmc5T-1


See screenshot to see issue: 

CleanShot 2023-07-31 at 17 17 09@2x

3. The small buttons that appear in the list items are missing hover states. 

CleanShot 2023-07-31 at 17 18 52@2x

Please see the hover states here: 
https://www.figma.com/file/zOI7oSH3fG1XDmmPzsP6PX/App-Navigation?type=design&node-id=4%3A16337&mode=design&t=24AXSMD60gHgmc5T-1
 
4. See VQA Issues Here: 
https://www.figma.com/file/zOI7oSH3fG1XDmmPzsP6PX/App-Navigation?type=design&node-id=655%3A63725&mode=design&t=24AXSMD60gHgmc5T-1

File: src/shell/components/NavTree/components/NavTreeItem.tsx [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+            }
+          >
+            {actions?.map((action) => {
+              return action;
+            })}
+          </Stack>
+        </Stack>
+      }
+      sx={{
+        "& .MuiTreeItem-content": {
```
[the remark this answers is not in the document]
Comment (theofficialnar) [reply src/shell/components/NavTree/components/NavTreeItem.tsx:68]: Adding these set of changes on the theme level will also affect the look of those components using the tree item already which I am not sure is a desired outcome that's why I just added these style changes locally.

File: src/apps/content-editor/src/app/components/ContentNav/index.tsx [hunk 2]
```diff
@@ -0,0 +1,786 @@
...
+                  onToggleCollapse={(paths) => setExpandedHiddenItems(paths)}
+                />
+              </AccordionDetails>
+            </Accordion>
+          </>
+        )}
+      </AppSideBar>
+      {isCreateContentDialogOpen && (
+        <CreateContentItemDialog
+          open
```
[the remark this answers is not in the document]
Comment (theofficialnar) [reply src/apps/content-editor/src/app/components/ContentNav/index.tsx:744]: It's a required prop on the component though that's why I added it initially. But I'll just change the component to make it optional with a default value of true.

File: src/apps/content-editor/src/app/components/ContentNav/index.tsx [hunk 3]
```diff
@@ -0,0 +1,786 @@
...
+            path = `/content/${navItem.ZUID}`;
+            break;
+        }
+
+        let actions = [
+          <IconButton
+            data-cy="tree-item-hide"
+            key="tree-item-hide"
+            sx={{
+              borderRadius: 0.5,
```
[the remark this answers is not in the document]
Comment (theofficialnar) [reply src/apps/content-editor/src/app/components/ContentNav/index.tsx:234]: Currently waiting on Zosh's feedback regarding this.

File: src/apps/content-editor/src/app/ContentEditor.js [hunk 4]
```diff
@@ -139,13 +137,13 @@ export default function ContentEditor() {
...
               )}
             >
-              <ContentNav
-                data-cy="contentNav"
-                dispatch={dispatch}
-                models={contentModels}
-                nav={navContent}
-              />
-
+              <ResizableContainer
```
[the remark this answers is not in the document]
Comment (theofficialnar) [reply src/apps/content-editor/src/app/ContentEditor.js:140]: Added on latest commit

File: src/apps/content-editor/src/app/components/ContentNav/components/NavError.tsx
[the remark this answers is not in the document]
Comment (theofficialnar) [reply src/apps/content-editor/src/app/components/ContentNav/components/NavError.tsx:1]: Updated, moved to root of folder

File: src/shell/components/AppSidebar/index.tsx [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+
+    return (
+      <ThemeProvider theme={themeMode}>
+        <ScopedCssBaseline
+          component={Box}
+          sx={{ height: "100%", width: "inherit" }}
+        >
+          <Stack
+            sx={{
+              backgroundColor: isLightMode ? "common.white" : "grey.900",
```
[the remark this answers is not in the document]
Comment (theofficialnar) [reply src/shell/components/AppSidebar/index.tsx]: Updated, added background.paper on dark theme pallete

File: src/shell/components/AppSidebar/index.tsx [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+  headerTitle?: string;
+}
+export const AppSideBar = forwardRef<any, PropsWithChildren<Props>>(
+  (
+    { mode = "light", HeaderSubComponent, headerTitle, children, ...props },
+    ref
+  ) => {
+    const childrenContainerRef = useRef<HTMLDivElement | null>(null);
+
+    const themeMode = useMemo(() => {
```
[the remark this answers is not in the document]
Comment (theofficialnar) [reply src/shell/components/AppSidebar/index.tsx]: Removed usememo

File: src/apps/content-editor/src/app/components/ContentNav/index.tsx [hunk 7]
```diff
@@ -0,0 +1,786 @@
...
+                    <IconButton
+                      onClick={() => {
+                        setCreateContentDialogLimit(["pageset", "templateset"]);
+                        setIsCreateContentDialogOpen(true);
+                      }}
+                      sx={{
+                        width: 20,
+                        height: 20,
+                        padding: 0.25,
+                        borderRadius: 0.5,
```
[the remark this answers is not in the document]
Comment (theofficialnar) [reply src/apps/content-editor/src/app/components/ContentNav/index.tsx:607]: Waiting for Zosh's feedback on the icon buttons

File: src/apps/content-editor/src/app/components/ContentNav/index.tsx [hunk 8]
```diff
@@ -0,0 +1,786 @@
...
+            navItem.label.toLowerCase()
+          )
+        ) {
+          actions.push(
+            <IconButton
+              data-cy="tree-item-add-new-content"
+              key="tree-item-add-new-content"
+              sx={{
+                borderRadius: 0.5,
+                backgroundColor: "primary.dark",
```
[the remark this answers is not in the document]
Comment (theofficialnar) [reply src/apps/content-editor/src/app/components/ContentNav/index.tsx]: Changed this to a regular button and color update will be done on theme

File: src/apps/content-editor/src/app/components/ContentNav/index.tsx [hunk 9]
```diff
@@ -0,0 +1,786 @@
...
+              >
+                <Button
+                  variant="contained"
+                  onClick={() => setIsCreateContentDialogOpen(true)}
+                  sx={{
+                    width: 24,
+                    height: 24,
+                    minWidth: 0,
+                    p: 0,
+                    backgroundColor: "primary.dark",
```
[the remark this answers is not in the document]
Comment (theofficialnar) [reply src/apps/content-editor/src/app/components/ContentNav/index.tsx]: Changed this to a regular button and color update will be done on theme

**Review by zcolah — CHANGES_REQUESTED** [review]
1. By default please keep all pages/items expanded by default instead of closed.

**Review by zcolah — CHANGES_REQUESTED** [review]
1. Randy has requested that we hide Releases for now as a tab in Content.

**Review by zcolah — CHANGES_REQUESTED** [review]
1. Please make the help tooltips show up after a delay of 1 second instead of 2 seconds

File: src/apps/content-editor/src/app/components/ContentNav/index.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (theofficialnar) [reply src/apps/content-editor/src/app/components/ContentNav/index.tsx]: Created a new icon button inside the @zesty-io/material-ui repo that is able to render a contained variant

File: src/apps/content-editor/src/app/components/ContentNav/index.tsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (theofficialnar) [reply src/apps/content-editor/src/app/components/ContentNav/index.tsx]: Created a new icon button inside the @zesty-io/material-ui repo that is able to render a contained variant

**Comment by theofficialnar** [comment]
@agalin920 I'll try to look for a way to optimize how the tree data is built. As for the NavTree being rendered 3 times, I don't think there's really any way around it since I had to put in 3 different NavTree components, each for Pages, Headless and Hidden items. 

I do agree that flattening out the filter list would make it faster since we'll only have to render 1 NavTree in that case but I think I still have to make some loops to add in the icons and the hover action elements for each nav item that matches the filter keyword.

**Pull request merged**